parfive/downloader.py

```diff
@@ -372,26 +375,120 @@ def callback(token, future, main_pb):
                                           total=get_http_size(resp))
                     else:
                         file_pb = None
-                    with open(str(filepath), 'wb') as fd:
-                        while True:
-                            chunk = await resp.content.read(chunksize)
-                            if not chunk:
-                                # Close the file progressbar
-                                if file_pb is not None:
-                                    file_pb.close()
+                    if max_splits and resp.headers.get('Accept-Ranges', None) == "bytes":
+                        # XXX: int(?)
+                        content_length = int(resp.headers['Content-length'])
+                        split_length = content_length//max_splits
+                        ranges = [
+                            [start, start + split_length]
+                            for start in range(0, content_length, split_length)
+                        ]
+                        # let the last part download everything
+                        ranges[-1][1] = ''
+                        queue = asyncio.Queue()
+                        workers = []
+                        for _range in ranges:
+                            workers.append(
+                                asyncio.create_task(Downloader._ranged_http_producer(
+                                    session, url, chunksize, _range, timeout, queue, **kwargs
+                                ))
+                            )
+                        consumer = asyncio.create_task(
```

Comment:
isn't this Python 3.7+ We still want to support 3.5 here, so I think you need `loop.create_class`

Reply:
> support 3.5

:(

Reply:
yeah, I am happy to drop 3.5 when it becomes a serious burden, but I don't think this counts :stuck_out_tongue: